Source/SwiftyDropbox/Shared/Handwritten/OAuth/AuthSession.swift

```diff
@@ -0,0 +1,104 @@
+///
+/// Copyright (c) 2020 Dropbox, Inc. All rights reserved.
+///
+
+import Foundation
+import CommonCrypto
+
+// MARK: Public
+
+/// Struct contains the information of a requested scopes.
+public struct ScopeRequest {
+    /// Type of the requested scopes.
+    public enum ScopeType: String {
+        case team
+        case user
+    }
+
+    let scopes: [String]
+    let includeGrantedScopes: Bool
+    let scopeType: ScopeType
+
+    var scopeString: String? {
+        guard !scopes.isEmpty else { return nil }
+        return scopes.joined(separator: " ")
+    }
+
+    /// Designated Initializer.
+    ///
+    /// - Parameters:
+    ///     - scopeType: Type of the requested scopes.
+    ///     - scopes: A list of scope returned by Dropbox server. Each scope correspond to a group of API endpoints.
+    ///       To call one API endpoint you have to obtains the scope first otherwise you will get HTTP 401.
+    ///     - includeGrantedScopes: If false, Dropbox will give you the scopes in scopes array.
+    ///       Otherwise Dropbox server will return a token with all scopes user previously granted your app
+    ///       together with the new scopes.
+    public init(scopeType: ScopeType, scopes: [String], includeGrantedScopes: Bool) {
+        self.scopeType = scopeType
+        self.scopes = scopes
+        self.includeGrantedScopes = includeGrantedScopes
+    }
+}
+
+// MARK: Internal
+
+/// Object that contains all the necessary data of an OAuth 2 Authorization Code Flow with PKCE.s
+struct AuthSession {
+    let scopeRequest: ScopeRequest?
+    let pkceData: PkceData
+    let state: String
+    let tokenAccessType = "offline"
+    let responseType = "code"
```

Comment:
please document the above non-private variables

Reply:
will do.